php/blocks/migration/class-post-type.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Post_Type.
+ *
+ * @package   Block_Lab
+ * @copyright Copyright(c) 2020, Block Lab
+ * @license http://opensource.org/licenses/GPL-2.0 GNU General Public License, version 2 (GPL-2.0)
+ */
+
+namespace Block_Lab\Blocks\Migration;
+
+use WP_Post;
+use WP_Query;
+
+/**
+ * Class Post_Type
+ */
+final class Post_Type {
+
+	/**
+	 * The previous slug of the custom post type (in Block Lab).
+	 *
+	 * @var string
+	 */
+	private $previous_post_type_slug;
+
+	/**
+	 * The new slug of the custom post type (not in Block Lab).
+	 *
+	 * @var string
+	 */
+	private $new_post_type_slug;
+
+	/**
+	 * The previous namespace of the block.
+	 *
+	 * @var string
+	 */
+	private $previous_block_namespace;
+
+	/**
+	 * The new namespace of the block.
+	 *
+	 * @var string
+	 */
+	private $new_block_namespace;
+
+	/**
+	 * The batch size (posts_per_page) for the post queries.
+	 *
+	 * @var int
+	 */
+	private $query_batch_size = 10;
+
+	/**
+	 * Post_Type constructor.
+	 *
+	 * @param string $new_post_type_slug  The new slug of the custom post type.
+	 * @param string $new_block_namespace The new namespace of the block.
+	 */
+	public function __construct( $new_post_type_slug, $new_block_namespace ) {
+		$this->previous_post_type_slug  = block_lab()->get_post_type_slug();
+		$this->new_post_type_slug       = $new_post_type_slug;
+		$this->previous_block_namespace = 'block-lab';
+		$this->new_block_namespace      = $new_block_namespace;
+	}
+
+	/**
+	 * Migrates all of the custom post type posts to the new slug and block namespace.
+	 *
+	 * These each store a config for a custom block,
+	 * they aren't actual posts as entered into the block editor.
+	 */
+	public function migrate_all() {
+		$offset = 0;
+		$posts  = $this->query_for_posts( $offset );
+
+		while ( ! empty( $posts ) ) {
+			foreach ( $posts as $post ) {
+				$this->migrate_single( $post );
```

Comment:
Should we break and return a `WP_Error` or something like that if this returns `false`?

Reply:
Good question, that might depend on the error handling in the UI.

If there are 25 `block_lab` posts, and only the 10th has an error in migration, should we keep migrating the other 15, or stop the whole migration and display an error in the UI?

I think the `block_lab` post migration is less prone to error than the normal `post` or `page` migration, but we won't know until it's used.

Reply:
That's a really good question. Let's leave this as it is now, so that it can continue if it hits an error, but keep that in mind as we develop the UI.

Reply:
Thanks, sounds good.

Reply:
If the migration of all of the `block_lab` posts failed, we'd definitely want some kind of feedback.